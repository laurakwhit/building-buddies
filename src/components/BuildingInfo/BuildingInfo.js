import React from 'react';
import PropTypes from 'prop-types';

import Nav from '../Nav/Nav';

import './BuildingInfo.scss';

const BuildingInfo = ({ name, address, handleLogOut }) => {
  return (
    <div className="building-info">
      <Nav handleLogOut={handleLogOut} />
      <div className="main-content">
        <div className="building-details">
          <h1>Building Info</h1>
          <h2>
            Name: <span>{name}</span>
          </h2>
          <h2>
            Address: <span>{address}</span>
          </h2>
        </div>
      </div>
    </div>
  );
};

BuildingInfo.propTypes = {
  name: PropTypes.string,
  address: PropTypes.string,
  handleLogOut: PropTypes.func
};

export default BuildingInfo;
