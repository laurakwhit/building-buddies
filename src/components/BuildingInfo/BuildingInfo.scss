.building-info {
  display: flex;
  .main-content {
    width: calc(100% - 20rem);
    color: #112229;
    margin: auto;
    h1 {
      margin: 0;
      font-size: 3rem;
    }
    h2 span {
      color: #76939e;
    }
    .building-details {
      margin: auto;
      width: 50%;
      text-align: left;
    }
  }
}
