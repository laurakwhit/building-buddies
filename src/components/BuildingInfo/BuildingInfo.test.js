import React from 'react';
import { shallow } from 'enzyme';

import BuildingInfo from './BuildingInfo';

describe('BuildingInfo component', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<BuildingInfo />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
