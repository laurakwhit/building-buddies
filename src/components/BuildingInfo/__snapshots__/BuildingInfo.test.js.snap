// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BuildingInfo component should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BuildingInfo />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Nav />,
        <div
          className="main-content"
        >
          <div
            className="building-details"
          >
            <h1>
              Building Info
            </h1>
            <h2>
              Name: 
              <span />
            </h2>
            <h2>
              Address: 
              <span />
            </h2>
          </div>
        </div>,
      ],
      "className": "building-info",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "handleLogOut": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="building-details"
          >
            <h1>
              Building Info
            </h1>
            <h2>
              Name: 
              <span />
            </h2>
            <h2>
              Address: 
              <span />
            </h2>
          </div>,
          "className": "main-content",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1>
                Building Info
              </h1>,
              <h2>
                Name: 
                <span />
              </h2>,
              <h2>
                Address: 
                <span />
              </h2>,
            ],
            "className": "building-details",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Building Info",
              },
              "ref": null,
              "rendered": "Building Info",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Name: ",
                  <span />,
                ],
              },
              "ref": null,
              "rendered": Array [
                "Name: ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "span",
                },
              ],
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Address: ",
                  <span />,
                ],
              },
              "ref": null,
              "rendered": Array [
                "Address: ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "span",
                },
              ],
              "type": "h2",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Nav />,
          <div
            className="main-content"
          >
            <div
              className="building-details"
            >
              <h1>
                Building Info
              </h1>
              <h2>
                Name: 
                <span />
              </h2>
              <h2>
                Address: 
                <span />
              </h2>
            </div>
          </div>,
        ],
        "className": "building-info",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "handleLogOut": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="building-details"
            >
              <h1>
                Building Info
              </h1>
              <h2>
                Name: 
                <span />
              </h2>
              <h2>
                Address: 
                <span />
              </h2>
            </div>,
            "className": "main-content",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h1>
                  Building Info
                </h1>,
                <h2>
                  Name: 
                  <span />
                </h2>,
                <h2>
                  Address: 
                  <span />
                </h2>,
              ],
              "className": "building-details",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Building Info",
                },
                "ref": null,
                "rendered": "Building Info",
                "type": "h1",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Name: ",
                    <span />,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Name: ",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": undefined,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "span",
                  },
                ],
                "type": "h2",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Address: ",
                    <span />,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Address: ",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": undefined,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "span",
                  },
                ],
                "type": "h2",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
